ballerina/build.gradle

```
@@ -1,38 +1,95 @@
 import org.apache.tools.ant.taskdefs.condition.Os
 
+buildscript {
+    repositories {
+        maven {
+            url = 'https://maven.pkg.github.com/ballerina-platform/plugin-gradle'
+            credentials {
+                username System.getenv("packageUser")
+                password System.getenv("packagePAT")
+            }
+        }
+    }
+    dependencies {
+        classpath "io.ballerina:plugin-gradle:${project.ballerinaGradlePluginVersion}"
+    }
+}
 
-/*
- * This file was generated by the Gradle 'init' task.
- */
-//apply plugin: 'io.ballerina.plugin'
 
-task build {
-    doLast {
-        project.exec {
-            if (Os.isFamily(Os.FAMILY_WINDOWS)){
-                commandLine 'cmd', '/c', 'bal pack && exit %%ERRORLEVEL%%'
-            }
-            else{
-                commandLine 'sh', '-c', 'bal pack'
-            }
+description = 'Ballerina - Pipe Package'
+
+def packageName = "pipe"
+def packageOrg = "nuvindu"
+def tomlVersion = stripBallerinaExtensionVersion("0.1.0")
+def ballerinaTomlFilePlaceHolder = new File("${project.rootDir}/build-config/resources/Ballerina.toml")
+def ballerinaTomlFile = new File("$project.projectDir/Ballerina.toml")
+
+def stripBallerinaExtensionVersion(String extVersion) {
+    if (extVersion.matches(project.ext.timestampedVersionRegex)) {
+        def splitVersion = extVersion.split('-')
+        if (splitVersion.length > 3) {
+            def strippedValues = splitVersion[0..-4]
+            return strippedValues.join('-')
+        } else {
+            return extVersion
         }
+    } else {
+        return extVersion.replace("${project.ext.snapshotVersion}", "")
     }
 }
-task publishToLocalRepo{
+
+apply plugin: 'io.ballerina.plugin'
+
+ballerina {
+    packageOrganization = packageOrg
+    module = packageName
+    langVersion = ballerinaLangVersion
+    testCoverageParam = "--code-coverage --coverage-format=xml"
```

Shall we add the `includes` flag as well?

And when looking at the native package naming, I suppose it should be changed to something like this:

`org.nuvindu.pipe.runtime.*`

Here the convention is to reverse the (imaginary) website name of the publisher (you), and then add separation for the special package and its modules.

```suggestion
    testCoverageParam = "--code-coverage --coverage-format=xml --includes=<your_native_package_path>.*:<your_ballerina_org>:<your_ballerina_package>*"
```